engine/src/main/java/org/terasology/world/block/BlockRegion.java

```diff
@@ -0,0 +1,243 @@
+    }
+
+    @Override
+    public BlockRegion setMax(int maxX, int maxY, int maxZ) {
+        return (BlockRegion) super.setMax(maxX, maxY, maxZ);
+    }
+
+    @Override
+    public BlockRegion setMin(int minX, int minY, int minZ) {
+        return (BlockRegion) super.setMin(minX, minY, minZ);
+    }
+
+    @Override
+    public BlockRegion union(AABBi other) {
+        return (BlockRegion) super.union(other);
+    }
+
+    @Override
+    public BlockRegion union(Vector3ic p) {
+        return (BlockRegion) super.union(p);
+    }
+
+    public BlockRegion union(Vector3ic p, BlockRegion dest) {
+        return (BlockRegion) super.union(p, dest);
+    }
+
+    public BlockRegion union(int x, int y, int z, BlockRegion dest) {
+        return (BlockRegion) super.union(x, y, z, dest);
+    }
+
+    public BlockRegion union(BlockRegion other) {
+        return (BlockRegion) super.union(other);
+    }
+
+    public BlockRegion union(AABBi other, BlockRegion dest) {
+        return (BlockRegion) super.union(other, dest);
+    }
+
+    @Override
+    public BlockRegion correctBounds() {
+        return (BlockRegion) super.correctBounds();
+    }
+
+    @Override
+    public BlockRegion union(int x, int y, int z) {
+        return (BlockRegion) super.union(x, y, z);
+    }
+
+    public BlockRegion setSize(int x, int y, int z) {
+        this.maxX = this.minX + x;
+        this.maxY = this.minY + y;
+        this.maxZ = this.minZ + z;
+        return this;
+    }
+
+    public BlockRegion setSize(Vector3ic size) {
+        return setSize(size.x(), size.y(), size.z());
+    }
+
+    public Vector3i getSize(Vector3i dest) {
+        return dest.set(this.maxX - this.minX, this.maxY - this.minY, this.maxZ - this.minZ);
+    }
+
+    public int getSizeX() {
+        return this.maxX - this.minX;
+    }
+
+    public int getSizeY() {
+        return this.maxY - this.minY;
+    }
+
+    public int getSizeZ() {
+        return this.maxZ - this.minZ;
+    }
+
+    public BlockRegion unionBlock(Vector3ic p) {
+        return unionBlock(p.x(), p.y(), p.z(), this);
+    }
+
+    public BlockRegion unionBlock(Vector3ic p, BlockRegion dest) {
+        return unionBlock(p.x(), p.y(), p.z(), dest);
+    }
+
+    @Override
+    public BlockRegion translate(int x, int y, int z) {
+        return (BlockRegion) super.translate(x, y, z);
+    }
+
+    public BlockRegion translate(Vector3ic xyz, BlockRegion dest) {
+        return (BlockRegion) super.translate(xyz, dest);
+    }
+
+    @Override
+    public BlockRegion translate(Vector3ic xyz) {
+        return (BlockRegion) super.translate(xyz);
+    }
+
+    @Override
+    public BlockRegion transform(Matrix4fc m) {
+        return (BlockRegion) super.transform(m);
+    }
+
+    public BlockRegion transform(Matrix4fc m, BlockRegion dest) {
+        return (BlockRegion) super.transform(m, dest);
+    }
+
+    public BlockRegion unionBlock(int x, int y, int z, BlockRegion dest) {
```

Comment:
unionBlock() adds +1 to the max value. I assume that is to make sure that the specified vector is fully enclosed into the bounding box. Is that correct? I'm wondering why the same thing isn't done for the min vector. If my region min is at (10, 10, 10) and I'm unionizing with (5, 5, 5), don't I expect the new min to be at (4, 4, 4)? Why would this only be relevant for max?
This method should also have some Javadoc on it.

Reply:
a block is from the corner to +1 so the max range is increased by 1. then it also makes it a bit funny when you setMax because in this case your are specific just the max corner of the AABB region and not the block.

Reply:
Is the purpose of this to emulate the `createFromMinMax()` behavior of `Region3i`? (because that was similar)
I don't really get why the definition of a block being "an AABB with max + 1" is useful at all.